mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -342,23 +343,24 @@ def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
             file_size = os.path.getsize(local_file)
             num_chunks = _compute_num_chunks(local_file, _MULTIPART_UPLOAD_CHUNK_SIZE)
             use_single_part_upload = num_chunks == 1
-            for index in range(num_chunks):
-                start_byte = index * _MULTIPART_UPLOAD_CHUNK_SIZE
-                future = self.thread_pool.submit(
-                    self._retryable_adls_function,
-                    func=patch_adls_file_upload,
-                    artifact_path=artifact_path,
-                    sas_url=credentials.signed_uri,
-                    local_file=local_file,
-                    start_byte=start_byte,
-                    size=_MULTIPART_UPLOAD_CHUNK_SIZE,
-                    position=start_byte,
-                    headers=headers,
-                    is_single=use_single_part_upload,
-                )
-                futures[future] = index
+            with self._create_thread_pool() as executor:
```

Comment:
@harupy If I understand correctly, we have a threadpool that enables multiple files to be uploaded at once, and, for each large file, we have another threadpool for uploading the large file in chunks. Since the recommended number of threads is `os.cpu_count() * 2`, aren't we potentially using too many threads here? How should we handle this?

Reply:
Would it be sufficient to have one threadpool for parallezing large file parts and one threadpool for parallelizing uploads across multiple files? E.g. "cross_file_upload_pool", "multi_part_file_upload_pool" or something?

Reply:
> have one threadpool for parallezing large file parts and one threadpool for parallelizing uploads across multiple files

This sounds good :) will update the code